omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskDefinitionResource1_8.java

```diff
@@ -0,0 +1,167 @@
+		this.taskServiceWrapper = taskServiceWrapper;
+	}
+	
+	@Override
+	public TaskDefinition newDelegate() {
+		return new TaskDefinition();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("taskClass");
+			description.addProperty("startTime");
+			description.addProperty("started");
+			description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("taskClass");
+			description.addProperty("startTime");
+			description.addProperty("lastExecutionTime");
+			description.addProperty("repeatInterval");
+			description.addProperty("startOnStartup");
+			description.addProperty("startTimePattern");
+			description.addProperty("started");
+			description.addProperty("properties");
+			description.addSelfLink();
+			description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("taskClass");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("name");
+		description.addRequiredProperty("description");
+		description.addRequiredProperty("taskClass");
+		description.addRequiredProperty("startTime");
+		description.addRequiredProperty("repeatInterval");
+		description.addRequiredProperty("startOnStartup");
+		description.addRequiredProperty("properties");
+		return description;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#save(java.lang.Object)
+	 */
+	@Override
+	public TaskDefinition save(TaskDefinition taskDefinition) {
+		taskServiceWrapper.saveTaskDefinition(taskDefinition);
+		return Context.getSchedulerService().getTaskByName(taskDefinition.getName());
```

Comment:
Why do you fetch the task again?

Reply:
No idea at that time, Just return taskDefinition is enough :-(